### PR TITLE
Fearture find shop by distance

### PR DESCRIPTION
**Pull request by phong-jack**
write feature find shop by distance
image

### CODE REVIEW

File: src/modules/shop/services/shop.service.ts [hunk 1]
```diff
@@ -27,4 +40,47 @@ export class ShopService {
...
+    );
+    const shopUpdated = await this.shopRepository.updateShopLocate(
+      id,
+      address,
+      locate.id,
+    );
+    return shopUpdated;
+  }
+
+  async findShopByDistance(userId: number) {
```
Comment (voxuanan) [inline src/modules/shop/services/shop.service.ts:63]: - I recommend you write a methods on repository doing this (you can using Query Builder, Raw Query),
- findShopByDistance should be paginated

File: src/modules/setting/constants/setting.constant.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export enum SETTING_KEY {
```
Comment (voxuanan) [inline src/modules/setting/constants/setting.constant.ts:1]: I see you prepare some potential "SETTING KEY" here, do we need to emit an event when user (shop) created to initialize the settings for them

File: src/modules/shop/repositories/shop.repository.ts [hunk 3]
```diff
@@ -14,6 +14,10 @@ export class ShopRepository {
     return await this.shopRepository.find();
   }
 
+  async findAllWithDistance(): Promise<Shop[]> {
+    return await this.shopRepository.find({ relations: { locate: true } });
+  }
+
```
Comment (voxuanan) [inline src/modules/shop/repositories/shop.repository.ts]: i recommend you to create base repositories (CRUD, find pagination) and extend it with each repository you have

File: src/modules/shop/services/shop.service.ts [hunk 4]
```diff
@@ -27,4 +40,47 @@ export class ShopService {
   async deleteShop(id: number) {
     await this.shopRepository.deleteShop(id);
   }
+
+  async updateShopLocate(id: number, address: string): Promise<Shop> {
```
Comment (voxuanan) [inline src/modules/shop/services/shop.service.ts:48]: What happens if I update the shop address using the above updateShop method, do we need to update the locate?

**Review by voxuanan — CHANGES_REQUESTED** [review]
Please update according to feedback

File: src/modules/shop/services/shop.service.ts (hunk 4 above)
Comment (phong-jack) [reply src/modules/shop/services/shop.service.ts:48]: image
import updateShopLocate into update Shop, remove route update-shop-locate

File: src/modules/shop/repositories/shop.repository.ts (hunk 3 above)
Comment (phong-jack) [reply src/modules/shop/repositories/shop.repository.ts]: apply CRUD base

File: src/modules/setting/constants/setting.constant.ts (hunk 2 above)
Comment (phong-jack) [reply src/modules/setting/constants/setting.constant.ts:1]: create event emitter for setting key

**Pull request merged**